packages/vite/src/node/optimizer/index.ts

```diff
@@ -47,9 +47,28 @@ export interface DepOptimizationOptions {
    */
   exclude?: string[]
   /**
-   * The bundler sometimes needs to rename symbols to avoid collisions.
-   * Set this to `true` to keep the `name` property on functions and classes.
-   * https://esbuild.github.io/api/#keep-names
+   * Options to pass to esbuild during the dep scanning and optimization
+   *
+   * - `entryPoints` are merged
+   * - `plugins` are merged with Vite's dep plugin
+   * - `external` takes precedence over `optimizeDeps.exclude`
+   * - `keepNames` takes precedence over the deprecated `optimizeDeps.keepNames`
+   *
+   * https://esbuild.github.io/api
+   */
+  esbuildOptions?: Omit<
+    EsbuildBuildOptions,
+    | 'bundle'
+    | 'write'
+    | 'watch'
+    | 'outdir'
+    | 'outfile'
+    | 'outbase'
+    | 'outExtension'
+    | 'metafile'
```

Comment:
Added this because I don't believe changing these would have a valid use case. Vite depends on bundling, the metafile, the output settings, and the build mode (i.e. watch = false).

Reply:
Should we add a comment within the code about why they are omitted?

Reply:
I like the inline documentation of what's omitted, but I added a comment.

e.g.:

![image](https://user-images.githubusercontent.com/10807310/116794696-4eaf5e00-aacf-11eb-8ecc-e9679193cdf9.png)